node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/ConfigServerApiImpl.java

```diff
@@ -202,24 +209,34 @@ private static CloseableHttpClient createClient(SSLConnectionSocketFactory socke
         cm.setMaxTotal(200); // Increase max total connections to 200, which should be enough
 
         // Have experienced hang in socket read, which may have been because of
-        // system defaults, therefore set explicit timeouts. Even though the Orchestrator
-        // server-side timeout is 10s, a 409 has been observed to be returned after ~30s
-        // in a case possibly involving zk leader election.
-        int timeoutMs = 30_000;
-        RequestConfig requestBuilder = RequestConfig.custom()
-                .setConnectTimeout(timeoutMs) // establishment of connection
-                .setConnectionRequestTimeout(timeoutMs) // connection from connection manager
-                .setSocketTimeout(timeoutMs) // waiting for data
+        // system defaults, therefore set explicit timeouts.
+        RequestConfig defaultRequestConfig = RequestConfig.custom()
+                .setConnectionRequestTimeout(1_000) // connection from connection manager
+                .setConnectTimeout(10_000) // establishment of connection
+                .setSocketTimeout(10_000) // waiting for data
                 .build();
 
         return HttpClientBuilder.create()
-                .setDefaultRequestConfig(requestBuilder)
+                .setDefaultRequestConfig(defaultRequestConfig)
                 .disableAutomaticRetries()
                 .setUserAgent("node-admin")
                 .setConnectionManager(cm)
                 .build();
     }
 
+    private static void setRequestConfigOverride(Params paramsOrNull, HttpRequestBase request) {
+        if (paramsOrNull == null) return;
+
+        RequestConfig.Builder builder = RequestConfig.copy(request.getConfig());
+
+        paramsOrNull.getConnectionTimeout().ifPresent(connectionTimeout -> {
+            builder.setConnectTimeout((int) connectionTimeout.getSeconds());
```

Comment:
Shouldn't this be millis?

Reply:
Great catch - I just verified this but forgot to update the code...